python/llm/src/bigdl/llm/ggml/convert_model.py

```diff
@@ -16,47 +16,99 @@
+                      "The input path {} was not a directory".format(input_path))
+    # shall we support model_id or just model directory?
+
     if dtype == 'int4':
         dtype = 'q4_0'
 
-    model_name = Path(input_path).stem
-    tmp_ggml_file_path = os.path.join(tmp_path, f'{model_name}_{int(time.time())}')
-    _convert_to_ggml(model_path=input_path,
-                     outfile_dir=tmp_ggml_file_path,
-                     model_family=model_family,
-                     outtype="fp16")
+    if tmp_path is not None:
+        model_name = Path(input_path).stem
+        tmp_ggml_file_path = os.path.join(tmp_path, f'{model_name}_{int(time.time())}')
+        _convert_to_ggml(model_path=input_path,
+                         outfile_dir=tmp_ggml_file_path,
+                         model_family=model_family,
+                         outtype="fp16")
+        tmp_ggml_file_path = next(Path(tmp_ggml_file_path).iterdir())
+        return quantize(input_path=tmp_ggml_file_path,
+                        output_dir=output_dir,
+                        model_family=model_family,
+                        dtype=dtype)
+    else:
+        with tempfile.TemporaryDirectory() as tmp_ggml_file_path:
+            _convert_to_ggml(model_path=input_path,
+                             outfile_dir=tmp_ggml_file_path,
+                             model_family=model_family,
+                             outtype="fp16")
+            tmp_ggml_file_path = next(Path(tmp_ggml_file_path).iterdir())
+            return quantize(input_path=tmp_ggml_file_path,
+                            output_dir=output_dir,
+                            model_family=model_family,
+                            dtype=dtype)
 
-    tmp_ggml_file_path = next(Path(tmp_ggml_file_path).iterdir())
 
-    return quantize(input_path=tmp_ggml_file_path,
-                    output_path=output_path,
-                    model_family=model_family,
-                    dtype=dtype)
+def main():
+    parser = argparse.ArgumentParser(description='Model Convert Parameters')
+    parser.add_argument('-i', '--input_path', type=str, required=True,
+                        help=("input_path, a path to a *directory* containing model weights"))
+    parser.add_argument('-o', '--output_dir', type=str, required=True,
+                        help=("output_dir,save path of output quantized model."))
+    parser.add_argument('-x', '--model_family', type=str, required=True,
+                        help=("model_family: Which model family your input model belongs to."
+                              "Now only `llama`/`bloom`/`gptneox` are supported."))
+    parser.add_argument('-t', '--dtype', type=str, default="int4",
+                        help="Which quantized precision will be converted.")
+    parser.add_argument('-p', '--tmp_path', type=str, default=None,
+                        help="Which path to store the intermediate model during the"
+                        "conversion process.")
```

Comment:
we need a `-h --help` for this cmd line tool in next PR. (Does `argparse` automatically generate one?)

Reply:
yeah `argparse` automatically generate one